test/functional/terminal/job_spec.lua

```diff
@@ -0,0 +1,71 @@
+local helpers = require('test.functional.helpers')(after_each)
+local thelpers = require('test.functional.terminal.helpers')
+local wait = helpers.wait
+
+describe('job handler', function()
+  local screen
+  local script_file = 'Xtest_job_4569.vim'
+  local input_file = 'test/functional/fixtures/job_spec_124KB.txt'
+
+  after_each(function()
+    os.remove(script_file)
+  end)
+
+  -- Note: This convoluted approach (spawned child nvim hosted in :terminal) is
+  -- the only way #4569 could be reproduced in a test. See also #4646.
+  function test_job_output_collection()
+    helpers.clear()
+    screen = thelpers.screen_setup(0, '["'..helpers.nvim_prog..
+    '", "-u", "NONE", "-i", "NONE", "--cmd", "set noswapfile"]')
+    screen:set_default_attr_ids({})
+    screen:set_default_attr_ignore(true)
+    screen.timeout = 15000
```

Comment:
This is a max of 15 seconds, but with the bugfix from #4646 the test succeeds in `.22` seconds (even with 3 retries).